File: v24_interpreter/java/SimpleInterpreter1.java
package v24_interpreter.java;

public class SimpleInterpreter1 {
    public static void main(String[] args) {
        String command = "REPEAT 3 TIMES: PRINT Hello";

        // Split the command into words based on whitespace
        String[] words = command.split(" ");

        //Handle the command
        if (words[0].equalsIgnoreCase("REPEAT")) {
            int repeatCount = Integer.parseInt(words[1]);
            for (int i = 0; i < repeatCount; i++) {
                if (words[3].equalsIgnoreCase("PRINT")) {
                    System.out.println(words[4]);
                }
            }
        }
    }
}

File: v24_interpreter/python/simple_interpreter1.py
def main():
    command = "REPEAT 3 TIMES: PRINT Hello"

    # Split the command into words based on whitespace
    words = command.split(" ")

    # Handle the command
    if words[0].upper() == "REPEAT":
        repeat_count = int(words[1])
        for i in range(repeat_count):
            if words[3].upper() == "PRINT":
                print(words[4])

if __name__ == "__main__":
    main()

